src/providers/postgres/qgsconnectionconfigurator.h

```diff
@@ -0,0 +1,180 @@
+      QgsDataSourceUri::SslMode sslmode = settings.enumValue( key + "/sslmode", QgsDataSourceUri::SslPrefer );
+
+      QString username;
+      QString password;
+      if ( settings.value( key + "/saveUsername" ).toString() == QLatin1String( "true" ) )
+      {
+        username = settings.value( key + "/username" ).toString();
+      }
+
+      if ( settings.value( key + "/savePassword" ).toString() == QLatin1String( "true" ) )
+      {
+        password = settings.value( key + "/password" ).toString();
+      }
+
+      // Old save setting
+      if ( settings.contains( key + "/save" ) )
+      {
+        username = settings.value( key + "/username" ).toString();
+
+        if ( settings.value( key + "/save" ).toString() == QLatin1String( "true" ) )
+        {
+          password = settings.value( key + "/password" ).toString();
+        }
+      }
+
+      QString authcfg = settings.value( key + "/authcfg" ).toString();
+
+      QgsDataSourceUri uri;
+      if ( !service.isEmpty() )
+      {
+        uri.setConnection( service, database, username, password, sslmode, authcfg );
+      }
+      else
+      {
+        uri.setConnection( host, port, database, username, password, sslmode, authcfg );
+      }
+      uri.setUseEstimatedMetadata( estimatedMetadata );
+
+      return uri;
+    }
+
+    static bool publicSchemaOnly( const QString &connName )
+    {
+      QgsSettings settings;
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/publicOnly", false ).toBool();
+    }
+
+    static bool geometryColumnsOnly( const QString &connName )
+    {
+      QgsSettings settings;
+
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/geometryColumnsOnly", false ).toBool();
+    }
+
+    static bool dontResolveType( const QString &connName )
+    {
+      QgsSettings settings;
+
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/dontResolveType", false ).toBool();
+    }
+
+    static bool useEstimatedMetadata( const QString &connName )
+    {
+      QgsSettings settings;
+
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/estimatedMetadata", false ).toBool();
+    }
+
+    static bool allowGeometrylessTables( const QString &connName )
+    {
+      QgsSettings settings;
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/allowGeometrylessTables", false ).toBool();
+    }
+
+    static bool allowProjectsInDatabase( const QString &connName )
+    {
+      QgsSettings settings;
+      return settings.value( "/" + T::mConnectionTypeName + "/connections/" + connName + "/projectsInDatabase", false ).toBool();
+    }
+
+    static void deleteConnection( const QString &connName )
+    {
+      QgsSettings settings;
+
+      QString key = "/" + T::mConnectionTypeName + "/connections/" + connName;
```

Comment:
I know it's boring but all string literals should be wrapped by QStringLiteral or QLatin1String.